exalearn-ui/src/app/components/new-listening/new-listening.component.ts

```diff
@@ -1,12 +1,143 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ElementRef, ViewChild } from '@angular/core';
+import { FormGroup, Validators, FormArray, FormGroupDirective, FormBuilder } from '@angular/forms';
+
+import { NewContentService } from '../../services/new-content.service';
+import { ApiService } from '../../services/api.service';
+import { AudioCloudService } from '../../services/audio-cloud.service';
+import { EnglishLevels } from '../../enums/enums';
+import { environment } from '../../../environments/environment.prod';
+import { API_PATH } from 'src/app/constants/api.constants';
 
 @Component({
 	selector: 'app-new-listening',
 	templateUrl: './new-listening.component.html',
 	styleUrls: ['./new-listening.component.scss']
 })
 export class NewListeningComponent implements OnInit {
-	constructor() {}
+	@ViewChild('fileInput')
+	fileInput: ElementRef;
+
+	@ViewChild(FormGroupDirective, { static: false })
+	listeningForm: FormGroupDirective;
+
+	form: FormGroup;
+	levels = Object.values(EnglishLevels);
+	percentage = 0;
+	url: string;
+	fileName: string;
+	loadAudio: boolean;
+	loadServer: boolean;
+	currentExercise = 0;
+	isURL: boolean;
+	notSelectedAnswers: string;
+
+	constructor(
+		private fb: FormBuilder,
+		private ncService: NewContentService,
+		private apiServise: ApiService,
+		private audioService: AudioCloudService
+	) {
+		this.form = this.fb.group({
+			level: ['', [Validators.required]],
+			exercises: this.fb.array([])
+		});
+	}
+
+	ngOnInit(): void {
+		this.ncService.generateListeningForm(this.form, environment.amountQuestions, environment.amountAnswers);
+	}
+
+	get exercises(): any {
```

Comment:
Can we replace this any by concrete type?

Reply:
Yes 🙈